cli/server.go

```diff
@@ -691,16 +691,17 @@ func Server(vip *viper.Viper, newAPI func(context.Context, *coderd.Options) (*co
 			}
 
 			client := codersdk.New(localURL)
-			if cfg.TLS.Enable.Value {
-				// Secure transport isn't needed for locally communicating!
+			if localURL.Scheme == "https" && isLocalhost(localURL.Hostname()) {
+				// The certificate will likely be self-signed or for a different
+				// hostname, so we need to skip verification.
```

Comment:
This comment confuses me a bit as to what the intent is here. If it is for a different hostname, we won’t skip verification, but the comment says otherwise?

Reply:
The coder server is only listening on TLS (not HTTP), but we still need to be able to make requests to the server further down in this function. If we don't disable TLS verification every request will most likely fail as most of the time the cert will be for `coder.corp.com` or something similar instead of `localhost`.

The `if` wrapper is just a sanity check. The host will always be `localhost` or equivalent in this URL, but in case we change that assumption in the future this `if` check will guard against security vulnerabilities.

Reply:
Ah, that makes sense. Thanks for the explanation! 👍